rocAL/rocAL/source/pipeline/tensor.cpp

```diff
@@ -320,17 +411,56 @@ unsigned rocalTensor::copy_data(hipStream_t stream, void *host_memory, bool sync
 unsigned rocalTensor::copy_data(void *user_buffer) {
     if (_mem_handle == nullptr) return 0;
 
+    #if ENABLE_HIP
+        if (_info._mem_type == RocalMemType::HIP) {
+            // copy from device to device
+            hipError_t status;
+            if ((status = hipMemcpyDtoD((void *)user_buffer, _mem_handle, _info.data_size())))
+                THROW("copy_data::hipMemcpyDtoH failed: " + TOSTR(status))
+        } else if (_info._mem_type == RocalMemType::HOST) {
+            // copy from host to device
+            hipError_t status;
+            if ((status = hipMemcpyHtoD((void *)user_buffer, _mem_handle, _info.data_size())))
+                THROW("copy_data::hipMemcpyHtoD failed: " + TOSTR(status))
+        }
+    #else
+        if (_info._mem_type == RocalMemType::HOST)
+            memcpy(user_buffer, _mem_handle, _info.data_size());
+    #endif
+
+    return 0;
+}
```

Comment:
```suggestion
    #if ENABLE_HIP
        if (_info._mem_type == RocalMemType::HIP) {
            // copy from device to device
            hipError_t status;
            if ((status = hipMemcpyDtoD((void *)user_buffer, _mem_handle, _info.data_size())))
                THROW("copy_data::hipMemcpyDtoH failed: " + TOSTR(status))
        } else if (_info._mem_type == RocalMemType::HOST) {
            // copy from host to device
            hipError_t status;
            if ((status = hipMemcpyHtoD((void *)user_buffer, _mem_handle, _info.data_size())))
                THROW("copy_data::hipMemcpyHtoD failed: " + TOSTR(status))
        }
    #else
        if (_info._mem_type == RocalMemType::HOST)
            memcpy(user_buffer, _mem_handle, _info.data_size());
    #endif

    return 0;
}
```

Don't add these changes in this PR